historical_system_profiles/views/v0.py

```diff
@@ -49,18 +40,51 @@ def _get_current_names_for_profiles(hsps):
     return enriched_hsps
 
 
+def _filter_old_hsps(hsps):
+    """
+    removes any HSPs with a captured_date older than now minus the valid profile age
+    """
+    now = datetime.datetime.now(datetime.timezone.utc)
+    cutoff = now - relativedelta(seconds=config.valid_profile_age_seconds)
+
+    valid_hsps = []
+    for hsp in hsps:
+        captured_date = dateutil.parser.parse(hsp.captured_date)
+        if captured_date > cutoff:
+            valid_hsps.append(hsp)
+
+    return valid_hsps
+
+
+def get_hsps_by_ids(profile_ids):
+    """
+    return a list of historical system profiles for the given profile IDs
+    """
+    account_number = view_helpers.get_account_number(request)
+
+    result = db_interface.get_hsps_by_profile_ids(profile_ids, account_number)
```

Comment:
You could go ahead and sort this from newest to oldest. I'm fine with a follow up to do that later.

Reply:
i can do it now, may as well

Reply:
actually, lets do this as a new PR. There is a related bug i'd have to fix at the same time where I'm exposing updated_on instead of captured_date.